File: books/TAoCP/v1/ch1_basic-concepts/code/euclid.py
def remainder(m, n):
    if m < n:
        return m
    return remainder(m - n, n)

def euclid(m, n):
    rem = remainder(m, n)
    if rem == 0:
        return n
    return euclid(n, rem)

if __name__ == "__main__":
    print("GCD(5,30) = %s" % euclid(5,30))
    print("GCD(100,29) = %s" % euclid(100,29))
    print("GCD(25,100) = %s" % euclid(25,100))
    print("GCD(123,33) = %s" % euclid(123,33))
    print("GCD(2166,6099) = %s" % euclid(2166,6099))
    print("GCD(6099,2166) = %s" % euclid(6099,2166))

File: books/TAoCP/v1/ch1_basic-concepts/code/euclid.clj
(ns ch1.euclid)

(defn mrem [m n]
  (if (< m n) m
      (mrem (- m n) n)))

(defn euclid [m n]
  (let [r (mrem m n)]
    (if (zero? r) n
        (euclid n r))))

(defn euclid-2 [m n]
  (if (zero? (mrem m n)) n
      (euclid (n (mrem m n)))))

(print "GCD(5,30) = " (euclid 5 30) "\n")
(print "GCD(100,29) = " (euclid 100 29) "\n")
(print "GCD(2,3) = " (euclid 2 3) "\n")
(print "GCD(25,100) = " (euclid 25 100) "\n")
(print "GCD(123,33) = " (euclid 123 33) "\n")
(print "GCD(2166,6099) = " (euclid 2166 6099) "\n")
(print "GCD(6099,2166) = " (euclid 6099 2166) "\n")
